test/solvers/advanced/forward_backward_test.py

```diff
@@ -0,0 +1,227 @@
+    prox_f = odl.solvers.proximal_zero(space)
+    grad_h = 2.0 * odl.IdentityOperator(space)  # Gradient of two-norm square
+
+    x = example_element(space)
+    x_global_min = space.zero()
+
+    forward_backward_pd(x, prox_f, prox_cc_g, lin_ops, grad_h, tau=0.5,
+                        sigma=[1.0], niter=10)
+
+    assert all_almost_equal(x, x_global_min, places=HIGH_ACCURACY)
+
+
+def test_forward_backward_with_lin_ops():
+    """Test for the forward-backward solver with linear operatros.
+
+    The test is done by minimizing ||x - b||_2^2 + ||alpha * x||_2^2. The
+    general problem is of the form
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``
+
+    and here we take f = 0, g = ||.||_2^2, L = alpha * IndentityOperator,
+    and h = ||. - b||_2^2.
+    """
+
+    space = odl.rn(10)
+    scaling = 0.1
+    b = example_element(space)
+
+    lin_op = scaling * odl.IdentityOperator(space)
+    lin_ops = [lin_op]
+    prox_cc_g = [odl.solvers.proximal_cconj_l2_squared(space)]
+    prox_f = odl.solvers.proximal_zero(space)
+    # Gradient of two-norm square
+    grad_h = 2.0 * (odl.IdentityOperator(space) - odl.ConstantOperator(b))
+
+    x = example_element(space)
+
+    # Explicit solution: x_hat = (I^T * I + (alpha*I)^T * (alpha*I))^-1 * (I*b)
+    op = odl.IdentityOperator(space) + lin_op.adjoint * lin_op
+    x_global_min = space.zero()
+    odl.solvers.conjugate_gradient(op, x_global_min, b, niter=10)
+
+    forward_backward_pd(x, prox_f, prox_cc_g, lin_ops, grad_h, tau=0.5,
+                        sigma=[1.0], niter=20)
+
+    assert all_almost_equal(x, x_global_min, places=LOW_ACCURACY)
+
+
+def test_forward_backward_with_li():
+    """Test for the forward-backward solver with infimal convolution.
+
+    The test is done by minimizing the functional ``(g @ l)(x)``, where
+
+        ``(g @ l)(x) = inf_y { g(y) + l(x-y) }``,
+
+    g is the indicator function on [-3, -1], and l(x) = 1/2||x||_2^2.
+    The optimal solution to this problem is given by x in [-3, -1].
+    """
+    # Parameter values for the box constraint
+    upper_lim = -1
+    lower_lim = -3
+
+    space = odl.rn(1)
+
+    lin_op = odl.IdentityOperator(space)
+    lin_ops = [lin_op]
+    prox_cc_g = [odl.solvers.proximal_cconj(
+                 odl.solvers.proximal_box_constraint(space,
+                                                     lower=lower_lim,
+                                                     upper=upper_lim))]
+    prox_f = odl.solvers.proximal_zero(space)
+    grad_h = odl.ZeroOperator(space)
+
+    grad_cc_ls = [odl.IdentityOperator(space)]
+
+    # Centering around a point further away from [-3,-1].
+    x = example_element(space) + space.element(3)
+    print(x)
+
+    forward_backward_pd(x, prox_f, prox_cc_g, lin_ops, grad_h, tau=0.5,
+                        sigma=[1.0], niter=20, grad_cc_l=grad_cc_ls)
+
+    solution_interv = odl.Interval(lower_lim - LOW_ACCURACY,
```

Comment:
A bit to complicated, simply do

```
assert lower_lim - LOW_ACCURACY < x
assert x < upper_lim + LOW_ACCURACY
```

this will also give better error messages on failure